### PR TITLE
Make sure package is installed before service started.

### PR DESCRIPTION
**Pull request by tedivm**
This is a simple change that makes sure the package is installed before the service itself is started. It only affects systems with "agent_install" set to true- if the puppet module does not install the package the service can still be managed without issue.

### CODE REVIEW

**Comment by smusali** [comment]
Hey @tedivm,

It looks like these changes are very interesting and necessary but unfortunately, it doesn't pass the tests. Can you make the appropriate changes in order to get this Pull Request pass the test?

Thanks in advance!

**Comment by tedivm** [comment]
Before I do that I want to make sure you understand the problem and the possible solutions. 

The problem is that the test suite was designed with the idea that each class would be completely independent of each other, but that meant that things like "ensure service is installed before it is started" was impossible to do.

That is why the tests are now failing- because they depend on each other as a complete project, rather than being a group of completely independent classes.

This leaves us with two options-

1. Erase all of the individual tests and focus on the test of the main class (init.pp, or `logdna`). That main class test also tests the other individual files, so this won't actually result in a loss of test coverage.

2. Rewrite the test suite completely and with the necessary bootstrapping code. This would potentially allow for more tests with different parameters but it is a lot of work (that I am personally not willing to do).

I also think there's a few things your team should be aware of-

1. The main "master" branch of this code doesn't work without this branch being merged. As I've mentioned in my previous prs (#1 and #2) the mainline branch of this repo has never actually worked.

2. There are two errors which are occurring that are not the result of my changes- specifically the `yumrepo` errors. This module doesn't have the `yum` project listed as a dependency in the `metadata.json` file, so the module is never installed. 

3. Your tests specifically allow (and indeed require) failure, so they're missing things. As I mentioned above there was an issue with a missing module that is still in the main repo- the reason your code isn't detecting that is you're (accidentally) suppressing the error. In this case it's because you're expecting a 'parse' error on non-redhat systems even though the file should parse just fine (it just shouldn't take any actions, but you're not actually testing actions *at all* in test suite anyways).


**TLDR:** My solution to fixing the tests is going to be to rip them out, because they were designed in a way that doesn't make any sense and in some cases they're actually hiding real errors.

**Comment by tedivm** [comment]
If you agree that ripping apart the test suite works for you I'll take care of it, and even add that missing module in for good measure. I just don't want to do it if it means you'll be less likely to accept the PR.

**Comment by tedivm** [comment]
I see you manually made one of the changes that this PR does, but without the other changes you're going to see errors still. It's not enough to remove the unused options- you need to define the relationships between the package and the service so that the package gets installed before the service does, and you need to force an apt upgrade before installing the package or puppet will fail on the first run (it may succeed on the second without this change but it's bad form to have to run puppet multiple times).

**Comment by smusali** [comment]
Hey @tedivm,

Thanks for your thoughts and opinions!

I will try to allocate time to work on the tests and respond with details. Most probably, this week I'll have time for this.

We appreciate your help!

**Comment by smusali** [comment]
@tedivm,

For now, these `rspec` tests are enough. We are going to revisit the tests later.

Now, I am merging this pull request in.

Thanks for all!
